app/src/main/java/com/training/movieapp/ui/main/MainActivity.kt

```diff
@@ -2,52 +2,46 @@ package com.training.movieapp.ui.main
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
-import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.NavigationUI.setupWithNavController
 import com.training.movieapp.R
 import com.training.movieapp.common.viewBinding
 import com.training.movieapp.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
 
     private val binding: ActivityMainBinding by viewBinding(ActivityMainBinding::inflate)
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
-        replaceFragment(MoviesFragment())
+        val navHostFragment = supportFragmentManager.findFragmentById(R.id.fragmentMain) as NavHostFragment
+        var navController = navHostFragment.navController
+        setupWithNavController(binding.bottomNavigation,navController)
         binding.apply {
             collapsingToolbar.title = "Movies"
             bottomNavigation.setOnItemSelectedListener {
                 when (it.itemId) {
-                    R.id.action_movies -> {
-                        replaceFragment(MoviesFragment())
+                    R.id.moviesFragment -> {
                         collapsingToolbar.title = "Movies"
+                        navController.navigate(R.id.moviesFragment)
                     }
-                    R.id.action_series -> {
-                        replaceFragment(SeriesFragment())
+                    R.id.seriesFragment -> {
                         collapsingToolbar.title = "Series"
+                        navController.navigate(R.id.seriesFragment)
                     }
-                    R.id.action_explore -> {
-                        replaceFragment(ExploreFragment())
+                    R.id.exploreFragment -> {
                         collapsingToolbar.title = "Explore"
+                        navController.navigate(R.id.exploreFragment)
                     }
-                    R.id.action_notification -> {
-                        replaceFragment(NotificationsFragment())
+                    R.id.notificationsFragment -> {
                         collapsingToolbar.title = "Notification"
+                        navController.navigate(R.id.notificationsFragment)
                     }
                     else -> {
-
                     }
                 }
                 true
             }
         }
```

Comment:
Chỗ này nếu e navigate kiểu này thì nó sẽ ko restore lại state của fragment nhỉ? Nếu muốn thay đổi title trên toolbar thì e nên dùng navController.addOnDestinationChangedListener để quan sát sự thay đổi màn hình rồi từ đó update title toolbar cho phù hợp. Note: setOnItemSelectedListener sẽ replace toàn bộ logic của thằng navigation component được gán cho thằng bottom navigation view sau khi gọi setupWithNavController. Còn addOnDestinationChangedListener sẽ add thêm 1 listener vô navController nên sẽ ko ảnh hưởng gì tới các listener của navController.

Reply:
Bản thân thằng setupWithNavController đã setup mọi thứ để việc navigate giữa các màn hình chính 1 cách rất thoải mái cho nên là hạn chế override lại logic của thằng setOnItemSelectedListener trừ khi bắt buộc phải làm nhoé